### ISSUE TITLE
Clang Static Analyzer "optin.cplusplus.UninitializedObject" false positive

### ISSUE THREAD

**Issue by tiagomacarios** [llvm/llvm-project#61919]
`optin.cplusplus.UninitializedObject` will flag the class `derived` below as having missing initializer. The Clang AST lists the field as being initialized. It looks like the initializer is missing logic for base class aggregate initialization.

https://godbolt.org/z/faaej1EP4
```cpp
struct base {
    bool b;
};

struct derived : base {
    int i;

    derived()
    : base{}
    , i{} // note: 1 uninitialized field at the end of the constructor call
    {} // warning: 1 uninitialized field at the end of the constructor call [clang-analyzer-optin.cplusplus.UninitializedObject]
};

void _() {
    derived d{}; // note: Calling default constructor for 'derived'
}
```

AST:
```
| |-CXXConstructorDecl <line:8:5, col:30> col:5 used derived 'void ()' implicit-inline
| | |-CXXCtorInitializer 'base':'base'
| | | `-InitListExpr <col:21, col:22> 'base':'base'
| | |   `-ImplicitValueInitExpr 'bool'
| | |-CXXCtorInitializer Field 0x557cf12f8dd8 'i' 'int'
| | | `-InitListExpr <col:26, col:27> 'int'
| | `-CompoundStmt <col:29, col:30>
```

**Comment by haoNoQ** [comment]
On-by-default checkers are affected by this as well! https://godbolt.org/z/M7M5hsn9v

Also an explicit initializer doesn't eliminate the issue! https://godbolt.org/z/8c6KzMs1b

Sounds serious.

@Szelethus this is also another situation where the uninitialized object checker splits the state unnecessarily in a race with VirtualCallChecker (but this isn't the root cause of this issue).

**Comment by steakhal** [comment]
Fixed by 39743580aff515ba47cb7f9ffb2b5c538cff91da
